paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -176,20 +184,25 @@ def get_services_to_k8s_namespaces(
 def sync_all_secrets(
     kube_client: KubeClient,
     cluster: str,
-    services_to_k8s_namespaces: Mapping[str, Set],
+    services_to_k8s_namespaces: Dict[str, Set],
     secret_provider_name: str,
-    vault_cluster_config: Mapping[str, str],
+    vault_cluster_config: Dict[str, str],
     soa_dir: str,
-    vault_token_file: str = DEFAULT_VAULT_TOKEN_FILE,
+    vault_token_file: str,
+    secret_type: Literal["all", "paasta-secret", "crypto-key", "boto-key"] = "all",
     overwrite_namespace: Optional[str] = None,
 ) -> bool:
     results = []
+
     for service, namespaces in services_to_k8s_namespaces.items():
+        sync_service_secrets: Dict[str, List[Callable]] = defaultdict(list)
```

Comment:
suggestion: 

thoughts on refactoring this a bit to something like:
```suggestion
        sync_service_secrets: Dict[str, Callable] = {
            'paasta-secret': partial(...),
            'boto-key': partial(...),
            'crypto-key': partial(...),
        }
        
        if secret_type == "all":
            result = (
                all(sync() for sync in sync_service_secrets.values())
            )
        else:
            result = sync_service_secrets[secret_type]()
        
        return result
```

so that we don't need to duplicate the secret syncing type key names in both the dict and the `all` case (+ simplify the result logic a bit)?

Reply:
`paasta-secret` type still needs to have a list of `Callable` because of namespaces. So, I am inclined to keep the previous changes.

Reply:
ah, missed that paasta secrets don't auto-discover the namespace from instance configs as the other ones do!